frontend/public/components/secret.jsx

```diff
@@ -58,7 +58,38 @@ const SecretDetails = ({obj: secret}) => {
 const SecretsList = props => <List {...props} Header={SecretHeader} Row={SecretRow} />;
 SecretsList.displayName = 'SecretsList';
 
-const SecretsPage = props => <ListPage ListComponent={SecretsList} canCreate={true} {...props} />;
+/* eslint-disable no-use-before-define */
+const secretType = (secret) => {
+  const { type } = secret;
+  const filter = _.head(filters);
+  if (!_.includes(filter.selected, type)) {
+    filter.selected.push(type);
+    filter.items.push({
+      'id': type,
+      'title': _.startsWith(type, 'kubernetes.io') ? _.replace(type, 'kubernetes.io/', '') : type
+    });
+  }
+  return type;
+};
+/* eslint-enable no-use-before-define */
```

Comment:
If you have to disable eslint's no-use-before-define, the code probably needs to change. Also, this code is modifying filters, but what's listening to changes on filters? I guess it just happens to work because something else causes a re-render.

I would strongly recommend against modifying filters here. It's fine to support just the hard-coded types in filters.items for now. Another PR can add support for arbitrary secret types.